dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumApplicationScope.kt

```diff
@@ -65,4 +89,69 @@ internal class RumApplicationScope(
     }
 
     // endregion
+
+    override fun onViewChanged(viewInfo: RumViewInfo) {
+        if (viewInfo.isActive) {
+            lastActiveViewInfo = viewInfo
+        }
+    }
+
+    @WorkerThread
+    private fun delegateToChildren(
+        event: RumRawEvent,
+        writer: DataWriter<Any>
+    ) {
+        val iterator = childScopes.iterator()
+        @Suppress("UnsafeThirdPartyFunctionCall") // next/remove can't fail: we checked hasNext
+        while (iterator.hasNext()) {
+            val result = iterator.next().handleEvent(event, writer)
+            if (result == null) {
+                iterator.remove()
+            }
+        }
+    }
+
+    @WorkerThread
+    private fun startNewSession(event: RumRawEvent, writer: DataWriter<Any>) {
+        val newSession = RumSessionScope(
+            this,
+            sdkCore,
+            samplingRate,
+            backgroundTrackingEnabled,
+            trackFrustrations,
+            this,
+            firstPartyHostHeaderTypeResolver,
+            cpuVitalMonitor,
+            memoryVitalMonitor,
+            frameRateVitalMonitor,
+            sessionListener,
+            contextProvider,
+            true
+        )
+        childScopes.add(newSession)
+        if (event !is RumRawEvent.StartView) {
+            lastActiveViewInfo?.let {
+                val key = it.keyRef.get()
+                if (key != null) {
+                    val startViewEvent = RumRawEvent.StartView(
+                        key = key,
+                        name = it.name,
+                        attributes = it.attributes
+                    )
+                    newSession.handleEvent(startViewEvent, writer)
+                } else {
+                    internalLogger.log(
+                        InternalLogger.Level.WARN,
+                        InternalLogger.Target.USER,
+                        LAST_ACTIVE_VIEW_GONE_WARNING_MESSAGE.format(it.name)
```

Comment:
minor: better to explicitly specify the `Locale`

```suggestion
                        LAST_ACTIVE_VIEW_GONE_WARNING_MESSAGE.format(Locale.US, it.name)
```